spec/system/reports/donations_summary_spec.rb

```diff
@@ -0,0 +1,166 @@
+RSpec.describe "DonationsSummary", type: :system, js: true do
+  before do
+    @url_prefix = "/#{@organization.short_name}"
+  end
+
+  context "When signed in as a normal user" do
+    before do
+      sign_in @user
+    end
+
+    let!(:storage_location) { create(:storage_location, :with_items, item_quantity: 0, organization: @organization) }
+
+    test_time = Time.zone.now
+
+    # 1, 20, 300, ..., 900000000
+    # assuming each value is used once, summing these values makes easily recognizable totals
+    # .fetch() from it so too-high indices raise IndexError
+    # legal indices are in -8..8 (i.e., inclusive)
+    item_quantities = (1..9).map { |i| i * 10**(i - 1) }
+
+    context "when visiting the summary page" do
+      it "has a link to create a new donation" do
+        visit @url_prefix + "/reports/donations_summary" 
+        expect(page).to have_link "New Donation", href: "#{@url_prefix}/donations/new"
+      end
+
+      # as of 28 Jan 2022, the "Recent Donations" list shows up to this many items matching the date filter
+      max_recent_donation_links_count = 3
+
+      # Make up to this many (inclusive) donations for each filtered period
+      # Keep it below (item_quantities.size - 1) so there's at least 2 values left for
+      # Donations outside of the filtered period
+      max_donations_in_filtered_period = max_recent_donation_links_count + 1
+
+      around do |example|
+        # Ensure "today" doesn't change for the server side during the example run
+        # Note, however, that this does *not* affect the client side
+        # So "today" for the server still might be "yesterday"
+        # —possibly even "last month" or "last year"-
+        # for the client
+        # Rely on rerun via rspec-retry for those edge cases
+        travel_to(test_time)
+        example.run
+        travel_back
```

Comment:
For a minute I got confused because I'm used to TimeCop where `travel` is different than `freeze`. But we're not using that in this project! That said, this could be slightly different:

```suggestion
        travel_to(test_time) do
          example.run
        end
```

... but also since I had to read the docs to learn how this isn't TimeCop I see that `travel_to` automatically adds a `travel_back`, so you could do a `before` instead of `around`.

All that said -- the way it's written works and I'd like to get this merged, so I might merge it.

Reply:
I just moved the existing spec and simplified it to avoid using a PageObject. I didn't mess with the time travel.